tests/test_pd_gmres.m

```diff
@@ -283,3 +283,51 @@ function test_size_compatibility_between_A_and_xInitial()
         assert(matches(ME.message, msg));
     end
 end
+
+function test_outputs_unrestarted_identity_matrix()
+    % Test whether the correct outputs are returned using an identity
+    % matrix when the unrestarted algorithm is called
+
+    % Setup a trivial linear system
+    A = eye(3);
+    b = ones(3, 1);
+
+    % Call function
+    [x, flag, relres, resvec, mvec, time] = pd_gmres(A, b);
+
+    % Compare with expected outputs
+    assertElementsAlmostEqual(x, ones(3, 1));
+    assert(flag == 1);
+    assertElementsAlmostEqual(relres, 0);
+    assertElementsAlmostEqual(resvec, [1.7320508075688770; 0]);
+    assert(isnan(mvec));
+    assert(time > 0 && time < 5);
+end
+
+function test_outputs_restarted_identity_matrix()
+    % Test whether the correct outputs are returned using an identity
+    % matrix when the restarted algorithm is called
+
+    A = eye(3);
+    b = [2; 3; 4];
+    mInitial = 1;
+    mMinMax = [1; 3];
+    mStep = 1;
+    tol = 1e-9;
+    maxit = 10;
+    xInitial = zeros(3, 1);
+    alphaPD = [-3; 5];
+
+    % Call function
+    [x, flag, relres, resvec, mvec, time] = ...
+        pd_gmres(A, b, mInitial, mMinMax, mStep, tol, maxit, xInitial, alphaPD);
+
+    % Compare with expected outputs
+    assertElementsAlmostEqual(x, [2; 3; 4]);
+    assert(flag == 1);
+    assert(relres <= tol);
+    assertElementsAlmostEqual(resvec, [1; 0]);
```

Comment:
This is ok, but remember that it will differ in matlab's gmres, since resvec is not a normalized vector.

Reply:
This should be fixed now. I will resolve the conversation.